src/Microsoft.ML.Featurizers/Common.cs

```diff
@@ -8,6 +8,16 @@
 
 namespace Microsoft.ML.Featurizers
 {
+    // Training state of the native featurizers
+    internal enum TrainingState
+    {
```

Comment:
Is it possible to have better names than Finished and Completed which are almost synonymous? How about enum TrainingState { NotStarted, InProgress, Completed, TransformerCreated}

Other suggestions for the final enum: ModelCreated, Finalized, ModelFinalized, etc.

Reply:
These are the same as the native side, so I think its best to leave them like this. I do agree though that the names are super close, so I will bring this up with my team.